File: src/main/java/org/railwaystations/rsapi/adapter/monitoring/LoggingMonitor.java
package org.railwaystations.rsapi.adapter.monitoring;

import org.railwaystations.rsapi.core.ports.Monitor;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;

import java.nio.file.Path;

@Service
@Profile("!prod")
public class LoggingMonitor implements Monitor {

    private static final Logger LOG = LoggerFactory.getLogger(LoggingMonitor.class);

    @Override
    public void sendMessage(final String message) {
        LOG.info(message);
    }

    @Override
    public void sendMessage(final String message, final Path file) {
        LOG.info(message + " - " + file);
    }

}

File: adapter/src/main/kotlin/org/railwaystations/rsapi/adapter/monitoring/LoggingMonitor.kt
package org.railwaystations.rsapi.adapter.monitoring

import org.railwaystations.rsapi.core.ports.outbound.MonitorPort
import org.railwaystations.rsapi.core.utils.Logger
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Service
import java.nio.file.Path

@Service
@ConditionalOnProperty(prefix = "monitor", name = ["service"], havingValue = "logging")
class LoggingMonitor : MonitorPort {

    private val log by Logger()

    override fun sendMessage(message: String) {
        log.info(message)
    }

    override fun sendMessage(message: String, file: Path?) {
        log.info("$message - $file")
    }
}
